### ISSUE TITLE
Does this package need a run_export?

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/gobject-introspection-feedstock#60]
### Comment:

Thank you the advise.

**Comment by pkgw** [comment]
I think maybe the current setup has what it needs? The `libgirepository` package doesn't have a run-export setting, but the `g-ir-host-tools` package has a run requirement on `libgirepository`, as does the `gobject-introspection` metapackage. Packages that are building with the gobject-introspection system will be depending on the host-tools subpackage in their host build, so that should introduce the dependency on the shared library at runtime. ... I think? Is there a specific issue that prompted this question?

**Comment by hmaarrfk** [comment]
I'm not sure that totally works. Users will have to manually add `libgirepository` to their run requirements.

The `run_export` guarantees that any addition to a `host` requirement is reflected as the appropriate runtime requirement (with correct pinnings).

**Comment by hmaarrfk** [comment]
Can you show me an example of a package that uses this feedstock with cross compilation. I'm having a hard time with it for libvips. It can help me answer your question better.

**Comment by pkgw** [comment]
[pango] does. So does [poppler]. Based on my grepping and searching it looks like most of our feedstocks just depend on the `gobject-introspection` kitchen-sink package and not the specific `g-ir-*` ones, so some of the run-exports might not be set up right.

[pango]: https://github.com/conda-forge/pango-feedstock
[poppler]: https://github.com/conda-forge/poppler-feedstock

**Comment by hmaarrfk** [comment]
It seems that there are 2 use cases:

1. the gobject introspection build tools as tools
2. The library.

Looking through the logs of pango and poppler, it seems that they only use the tools, and not the dynamic library at runtime.

libvips on the otherhand seems to need the library.  It may be that `libgirepository` needs to `run_export` itself.

**Comment by pkgw** [comment]
Hmm, yeah, if a package is providing binding data so that scripting languages can interact with it, I don't think it should need the library at runtime. That should only come in if a package is actually loading bindings and interacting with them. But regardless of all that, I have trouble seeing how a `run_export` in `libgirepository` would be *wrong*.

**Comment by ryanvolz** [comment]
> Hmm, yeah, if a package is providing binding data so that scripting languages can interact with it, I don't think it should need the library at runtime. That should only come in if a package is actually loading bindings and interacting with them. But regardless of all that, I have trouble seeing how a `run_export` in `libgirepository` would be _wrong_.

I agree, it seems correct that a package building against `libgirepository` surgically would benefit from a `run_export`. I just don't think we've encountered such a package until now.

**Comment by hmaarrfk** [comment]
Yeah I was kinda wondering the same about why it wasn't an error before. I just don't think anybody actually linked against libgirepository, the two packageds mentioned by pkgw don't list it as being used.
